src/ruby/puppetserver-lib/puppet/server/master.rb

```diff
@@ -102,6 +102,14 @@ def terminate
     Puppet::Server::Config.terminate_puppet_server
   end
 
+  def getTasks(env)
+    environment = @env_loader.get(env)
+    unless environment.nil?
+      tasks_per_env =
+        Puppet::InfoService.tasks_per_environment(environment.name)
+    end
```

Comment:
If no one wants to discuss the nil-if-env-doesn't-exist pattern, i'll just remove the assignment

Reply:
IMO its fine as-is. We're not going to let a request through the indirector for an environment that doesn't exist anyway. Might be worth adding a yard tag on the method though noting that it will return nil in this case.

Reply:
A yard tag? Is yard generated for Puppet server?

Also, I don't think the Ruby method returns nil, I think it might throw? Which is why the unless guard exists in the other code.

Reply:
Sorry - yes I'm confident we don't actually do anything with the tags in puppet server. A generic comment is fine. I was just thinking I wouldn't do any sort of cognitive double-take if I saw something like `# @return [Array, nil] an array of hashes describing tasks` here. 

And yes exactly - at the base level the implementation of `tasks_per_environment` will raise an exception if the environment doesn't exist because of `get!` here: https://github.com/puppetlabs/puppet/blob/master/lib/puppet/info_service/task_information_service.rb#L6